### DISCUSSION TITLE
NetworkX Guidance regarding GSOC 2021

### DISCUSSION THREAD

**Discussion by vadyushkins** [networkx/networkx#4689] [Q&A]
Hi there!

My name Vadim Abzalov and I am 3rd-year student at the faculty of Mathematics and Mechanics at Saint Petersburg State University in Russia. My specialization is "Software engineering".

I'm familiar with Python and graph theory, and I'm interested in deepening my understanding of graph algorithms and sharing them with the community.

I have been maintaining the **[CFPQ_Data](https://github.com/JetBrains-Research/CFPQ_Data)** and **[CFPQ_PyAlgo](https://github.com/JetBrains-Research/CFPQ_PyAlgo)** projects in **JetBrains Research** for a year now. These projects are entirely written in Python and are used to experimentally investigate graph algorithms in the [CFPQ](https://openproceedings.org/2021/conf/edbt/p48.pdf) area. Also in Python I wrote several algorithms that were included in the paper accepted at [EDBT: 24th International Conference on Extending Database Technology](https://edbticdt2021.cs.ucy.ac.cy/).

At the university, I passed the courses of discrete mathematics and graph theory with excellent results. I also took the [Graph Theory course at Coursera](https://coursera.org/share/2be803d4902cc4300e007684464dc0e2). At the moment at the university I am studying graph analysis algorithms.

That's why I was extremely interested in the project "Pedagogical Interactive Notebooks for Algorithms Implemented in NetworkX". I would like to gain a deeper understanding of the standard algorithms of network science and graph theory and help anyone interested to delve deeper into this topic with me. I think we could write a whole interactive course using these notebooks and this is extremely useful for anyone who wants to understand graph algorithms.

I am also interested in implementing algorithms mentioned in projects such as "Community Detection Algorithms" and "Directed Version of Traveling Salesman Problem". These algorithms are interesting to me by themselves and I would be happy to understand them deeply and write a detailed notebook about them.

I would be glad to get a mentor with whom we can discuss the possibilities of participating in these projects.

**Comment by vadyushkins** [comment]
Hi @MridulS!

To familiarize myself with NetworkX for GSOC 2021, I would like to work on [this issue](https://github.com/networkx/networkx/issues/3063).

Can you please tell me if this is a good first step?

**Reply by vadyushkins** [reply]
I opened a [PR](https://github.com/networkx/networkx/pull/4700) and will be glad to receive your comments!

**Comment by vadyushkins** [comment]
I finally decided that I wanted to work on project "Educational Interactive Notebooks for Algorithms Implemented in NetworkX" during GSOC 2021.

In order to familiarize myself with the process of writing notebooks before the official GSOC 2021 coding time, I would like to contribute some notebooks.

I am going to start from notebook about [Cycles](https://networkx.org/documentation/latest/reference/algorithms/dag.html) and [DAGs](https://networkx.org/documentation/latest/reference/algorithms/dag.html).

Could you tell me if this is a good first step to become a participant in this project during GSOC 2021?

Also if you have any ideas on the order of writing notebooks it would be helpful for me.

**Reply by Hetarth02** [reply]
Hello, I am also interested in the same project as you are for the GSOC 2021.

I already have tried to contribute some notebooks before the GSOC. If you want to check them, [here.](https://github.com/networkx/notebooks/pull/19)

The topics I have chosen were [Bipartite(Basic Functions)](https://networkx.org/documentation/stable/reference/algorithms/bipartite.html#module-networkx.algorithms.bipartite.basic) and [Matching.](https://networkx.org/documentation/stable/reference/algorithms/approximation.html#module-networkx.algorithms.approximation.matching)

If you have any suggestions to make the notebooks better, feel free to guide me.

[Networkx Notebooks](https://github.com/networkx/notebooks) is the place for submitting the notebooks.
